trio/_tests/test_exports.py

```diff
@@ -118,18 +116,45 @@ def no_underscores(symbols):
         )
         from mypy.api import run
 
-        res = run(["--config-file=", "--follow-imports=silent", str(tmpfile)])
-
-        # clean up created py.typed file
-        if not py_typed_exists:  # pragma: no cover
-            py_typed_path.unlink()
+        mypy_res = run(["--config-file=", "--follow-imports=silent", str(tmpfile)])
 
         # check that there were no errors (exit code 0), otherwise print the errors
-        assert res[2] == 0, res[0]
+        assert mypy_res[2] == 0, mypy_res[0]
         return
+    elif tool == "pyright_verifytypes":
+        import subprocess
+
+        res = subprocess.run(
+            ["pyright", f"--verifytypes={modname}", "--verbose"],
```

Comment:
Does `--outputjson` not require we parse the output like this?

Reply:
you can't combine `--outputjson` and `--verbose` (for some reason), and `--outputjson` won't include symbols without an error. So we have to do the dirty output parsing.
Adding a comment

Reply:
wait no, I think `--outputjson` does include symbols without error and I just got confused by the `trio.socket` shenanigans